### ISSUE TITLE
Bool are int

### ISSUE THREAD

**Issue by ChristophePapazian** [microsoft/pylance-release#2865]
## Environment data

-   Language Server version: 2022.5.3
-   OS and version: macOS 10.14.6 (Mojave)
-   Python version (& distribution if applicable, e.g. Anaconda): 3.10.4

## Code Snippet

```python
def test(N:int) -> bool:
  from random import choice
  return sum(a for a in [choice([True,False]) for _ in range(N)])==2
```

## Expected behavior

This code should be perfectly fine.

## Actual behavior

Problem detected on the last line
Pylance computes that it will always return False. Because it implies that sum on bool is always 0 or a bool. 

Condition will always evaluate to False since the types "bool | Literal[0]" and "Literal[2]" have no overlapPylance[reportUnnecessaryComparison](https://github.com/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md#diagnostic-severity-rules)

Two problems here : 
1/ bool should be considered as int, because they are ! (True+True==2)
2/ Even with ==1 at the end of the line, pylance seems to think that (True==1) is False.

**Comment by erictraut** [comment]
I'm glad to see that you're using the `reportUnnecessaryComparison` diagnostic check! It's a strict-mode check, so most pylance users are not aware of it.

This looks like a bug in the type definition of `sum`, which is defined in the typeshed stubs [here](https://github.com/python/typeshed/blob/1eef7365b9f68f6af023da731369096f55437654/stdlib/builtins.pyi#L1552):

```python
def sum(__iterable: Iterable[_SumT]) -> _SumT | Literal[0]: ...
```

That leads to the following:

```python
x = sum([True, True, True])
reveal_type(x) # bool | Literal[0]
```

However, the correct (runtime) type of this expression is `int`.

So the bug will need to be fixed in the typeshed stubs. It will probably require a special-case overload for `sum` to properly model its behavior. Please file a bug (or a PR) in the typeshed project.

Here are a few suggested workarounds:
1. Use ints rather than bools in your code:

```python
def test(N: int) -> bool:
    return sum([choice([1, 0]) for _ in range(N)]) == 2
```

2. Add a `# pyright: ignore[reportUnnecessaryComparison]` or `# type: ignore` comment to the end of the line.
3. Disable that diagnostic check for the file by adding a `# pyright: reportUnnecessaryComparison=false` comment at the top of the file.

**Comment by ChristophePapazian** [comment]
Thanks for the answer Eric.
Reported on [https://github.com/python/typeshed/issues/7974](https://github.com/python/typeshed/issues/7974)

**Comment by erictraut** [comment]
Thanks to the maintainers of typeshed, a fix was already integrated. I merged the latest typeshed stubs into pyright, so the next release of pylance will include them as well. I confirmed that the above issue is now fixed.

**Comment by bschnurr** [comment]
This issue has been fixed in version 2022.6.0, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202260-1-June-2022)
